packages/synapse-react-client/src/components/DataGrid/useDataGridWebsocket.ts

```diff
@@ -70,9 +121,17 @@ export function useDataGridWebSocket() {
     shouldEstablishWebsocketConnection,
     connectionParams,
     establishWebsocketConnection,
-    isVisible,
+    websocketOptionsWithoutModel,
   ])
 
+  /**
+   * Establish the WebSocket connection when conditions are met.
+   * Delegates to the mutation hook for presigned URL fetch, WebSocket creation, and retry logic.
+   */
+  useEffect(() => {
+    establishConnectionCallback()
+  }, [establishConnectionCallback])
```

Comment:
This is kind of an awkward pattern since it obscures which dependencies will trigger a connection. 

A nice medium might be to define the function in the effect:
```suggestion
  /**
   * Establish the WebSocket connection when conditions are met.
   * Delegates to the mutation hook for presigned URL fetch, WebSocket creation, and retry logic.
   */
  useEffect(() => {
    function establishConnection() {
      // ...
    }
    establishConnection()
  }, [
    shouldEstablishWebsocketConnection,
    connectionParams,
    establishWebsocketConnection,
    websocketOptionsWithoutModel
  ])
```